cdap-distributions/bin/prep_and_stage_bundles.sh

```diff
@@ -0,0 +1,136 @@
+
+if [ $# -lt 2 ]; then
+  die "You need to pass project and version to ${_script}"$'\n\n'"Example: ./${_script} cdap 3.0.0"
+fi
+
+#########################################################################
+create_bundles() {
+
+   ### PREP
+   # make sure bundle directories exist
+   mkdir -p ${STG_DIR} || die "Unable to create STG_DIR"
+   mkdir -p ${DEB_BUNDLE_DIR} ${RPM_BUNDLE_DIR} || die "Unable to create bundle processing directories"
+   mkdir -p ${STG_DEB_DIR} ${STG_RPM_DIR} || die "Unable to create bundle staging directories"
+   OUTGOING_DIR=${BUILD_PACKAGE}/${_version}
+   echo ${OUTGOING_DIR}
+   
+   ### FIND ARTIFACTS
+   mkdir -p ${S3STG_INCOMING_PROJ} || die "Unable to create project staging directory"
+   cd ${S3STG_INCOMING_PROJ} || die "Unable to cd to ${S3STG_INCOMING_PROJ} directory"
+   find . -name "${BUILD_PACKAGE}-*${VERSION}-*.deb" -exec cp '{}' ${DEB_BUNDLE_DIR}/. \; || die "Unable to copy Debian ${VERSION} packages to TMP directory"
+   find . -name "${BUILD_PACKAGE}-*${VERSION}-*.rpm" -exec cp '{}' ${RPM_BUNDLE_DIR}/. \; || die "Unable to copy ${VERSION} RPMs to TMP directory"
+   
+   # create and verify debian tarball
+   cd ${DEB_BUNDLE_DIR} || die "Unable to cd to ${DEB_BUNDLE_DIR} directory"
+   tar czvf ${DEB_BUNDLE_TGZ} * || die "Unable to create Debian tar bundle"
+   tar tzvf ${DEB_BUNDLE_TGZ} || die "Unable to show contents of Debian tar bundle"
+   
+   # create and verify RPM tarball
+   cd ${RPM_BUNDLE_DIR} || die "Unable to cd to ${RPM_BUNDLE_DIR} directory"
+   tar czvf ${RPM_BUNDLE_TGZ} * || die "Unable to create RPM tar bundle"
+   tar tzvf ${RPM_BUNDLE_TGZ} || die "Unable to show contents of RPM tar bundle"
+   
+   # SYNC bundle(s) to remote server
+   decho "rsyncing with rsync -av -e \"ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null\" ${RSYNC_QUIET} ${DEB_BUNDLE_TGZ_PATH} ${STG_DEB_DIR}/.  2>&1"
+   decho "rsyncing with rsync -av -e \"ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null\" ${RSYNC_QUIET} ${RPM_BUNDLE_TGZ_PATH} ${STG_RPM_DIR}/.  2>&1"
+   rsync -av -e "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" ${RSYNC_QUIET} ${DEB_BUNDLE_TGZ_PATH} ${STG_DEB_DIR}/. 2>&1 || die "could not rsync ${DEB_BUNDLE_TGZ_PATH} as ${REMOTE_USER} to ${REMOTE_HOST}: ${!}"
+   rsync -av -e "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" ${RSYNC_QUIET} ${RPM_BUNDLE_TGZ_PATH} ${STG_RPM_DIR}/. 2>&1 || die "could not rsync ${RPM_BUNDLE_TGZ_PATH} as ${REMOTE_USER} to ${REMOTE_HOST}: ${!}"
+
+   # clean up
+   #clean
+}
+
+######################################################################################
+decho "STARTING"
+
+# Check number of arguments. If <5 are passed, print help and exit.
+NUMARGS=$#
+decho -e \\n"Number of arguments: ${NUMARGS}"
+if [ ${NUMARGS} -lt 5 ]; then
+  HELP
+fi
+
+decho "#######################################################################################"
+echo "Creating RPM and Debian bundles"
+create_bundles
```

Comment:
Now that we've gotten here, `create_bundles` (and your new `sync_bundles` function) should both be simply included into `build_artifact_upload.sh` to DRY up the code and not repeat variable definitions, functions, etc.